src/SmoFacade/Server.cs

```diff
@@ -281,5 +281,23 @@ public void AddRole(LoginProperties login, RoleProperties role)
 
       matchingLogin.AddRole(role.Name);
     }
+
+    public void CheckDBConnection(string dbName, int connectionTimeout)
+    {
+      // Make sure to set `PersistSecurityInfo=True` in the connection string otherwise the password parameter might not be present
+      // https://docs.microsoft.com/en-us/dotnet/api/system.data.sqlclient.sqlconnection.connectionstring?view=dotnet-plat-ext-5.0#remarks

```

Comment:
I'd suggest that it might be better to put this comment before the method definition, which will generate a doc comment https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/documentation-comments

This is also kinda tricky because the `PersistSecurityInfo=True` needs to be passed into the AgDatabase constructor, not method.

It might be worth saving a `SqlConnectionStringBuilder` object as a member in AgDatabase class

Reply:
ah true, and then generate the connection string when needed
Let me look into it

Reply:
I've added in a private member variable to store the read-only connection string - to make sure no one changes it.
It can be used to create `SqlConnectionStringBuilder`s if someone needs a modified connection sting.

Hope that's fine, let me know your thoughts